inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go

```diff
@@ -129,9 +129,9 @@ func (h *heartbeatManager) consumerHB2Master() {
 				return
 			}
 		}
+		h.consumer.masterHBRetry = 0
```

Comment:
If HBRetry over max tims， need use other master to retry connect. 
But HBRetry over max cann't to change, if the master is downoff.

Reply:
@TszKitLo40, how about this problem?

Reply:
In https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L566
`master_sh_retry_cnt_` in C++ SDK is the same meaning of `masterHBRetry` in Go SDK. But it seems that C++ SDK also does not try to reconnect to another address if `master_sh_retry_cnt_` exceeds the `config_.GetMaxHeartBeatRetryTimes`, it just enlarge the value. Or do I miss some logic of C++ SDK? @gosonzhang

Reply:
@TszKitLo40, there should be a logical omission:
1. The master address obtained by each heartbeat is performed through getCurrentMasterAddr()[1];
2. In getCurrentMasterAddr(), according to the valid master address obtained during registration [2], locate the heartbeat through the curr_master_addr_ parameter;
3. The curr_master_addr_ value is effectively determined through the registration between the client and the Master [3]

1. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L510
2. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L1414
3. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L374

Reply:
![image](https://user-images.githubusercontent.com/14038849/182006966-9d8b38f7-fa13-496a-8fdc-735cfe2dcbb9.png)

In go, miss this logic